src/index.ts

```diff
@@ -1,164 +1,616 @@
+ * which then in turn fetches the data from the depot in your app.
+ *
+ * So if you want to build a service independent of Fission's infrastructure,
+ * you will need to write your own reference and depot implementations (see source code).
+ *
+ * NOTE: This uses all the default components as the dependencies for the auth component.
+ *       If you're, for example, using a non-default storage component, you'll want to
+ *       pass that in here as a parameter as well.
+ *
+ *       Dependents: crypto, manners, reference, storage.
+ */
+export const auth = {
+  /**
+   * A standalone authentication system that uses the browser's Web Crypto API
+   * to create an identity based on a RSA key-pair.
+   *
+   * NOTE: This uses a Fission server to register an account (DID).
+   *       Check out the `wnfs` and `base` auth implementations if
+   *       you want to build something without the Fission infrastructure.
+   */
+  async fissionWebCrypto(
+    config: Configuration,
+    options: {
+      disableWnfs?: boolean
+      staging?: boolean
+
+      // Dependents
+      crypto?: Crypto.Implementation
+      manners?: Manners.Implementation
+      reference?: Reference.Implementation
+      storage?: Storage.Implementation
+    } = {}
+  ): Promise<Auth.Implementation<Components>> {
+    const { disableWnfs, staging } = options
+
+    const manners = options.manners || defaultMannersComponent(config)
+    const crypto = options.crypto || await defaultCryptoComponent(config.namespace)
+    const storage = options.storage || defaultStorageComponent(config.namespace)
+    const reference = options.reference || await defaultReferenceComponent({ crypto, manners, storage })
+
+    if (disableWnfs) {
+      if (staging) return FissionAuthBaseStaging.implementation({ crypto, reference, storage })
+      return FissionAuthBaseProduction.implementation({ crypto, reference, storage })
     } else {
-      const fs = options.loadFileSystem === false ?
-        undefined :
-        await loadFileSystem(rootPermissions, authedUsername)
-
-      return appState.scenarioAuthed(
-        authedUsername,
-        fs
-      )
+      if (staging) return FissionAuthWnfsStaging.implementation({ crypto, reference, storage })
+      return FissionAuthWnfsProduction.implementation({ crypto, reference, storage })
     }
+  }
+}
 
-  } else if (authedUsername) {
-    return appState.scenarioAuthed(
-      authedUsername,
-      undefined
-    )
-
-  } else {
-    return appState.scenarioNotAuthed()
+/**
+ * Predefined confidences configurations.
+ *
+ * If you want partial read and/or write access to the filesystem you'll want
+ * a "confidences" component. This component is responsible for requesting
+ * and receiving UCANs, read keys and namefilters from other sources to enable this.
+ *
+ * NOTE: This uses all the default components as the dependencies for the confidences component.
```

Comment:
Hmm was about to make a similar rewrite suggestion as the one for the auth component above, but not so sure. Might be alright as is. 🤔

Reply:
Yeah I'm not entirely happy with this line either.
This gives the assumption that you can't customise it.

Maybe we just drop this line entirely and say:
> If you're using a non-default component, you'll want to pass that in here as a parameter as well.
> Dependents: crypto, manners, reference, storage.

Reply:
Yeah, that reads better to me! 👍